bigbluebutton-html5/imports/ui/components/audio/device-selector/component.jsx

```diff
@@ -70,13 +69,13 @@ class DeviceSelector extends Component {
         value={value}
         onChange={this.handleSelectChange}
         disabled={!options.length}
-        className={styles.select}
+        className={cx(styles.select, className)}
       >
         {
           options.length ?
-            options.map((option, i) => (
+            options.map(option => (
               <option
-                key={i}
+                key={_.uniqueId('device-option-')}
```

Comment:
We have now issues with this approach in the user list.
The problem is that it generates a new key every time the component is updated. Which in its turn makes React think that it's a completely new element, not an updated one.
The right approach is to generate keys in componentDidMount (just once) and then just use them in render.

Reply:
What exactly is the issue with react thinking it is a new element? 
Also, this component props will not be updated that often, so I fail to see the issue with the approach here.

Reply:
I guess it's not very important in this case, since it returns a simple <option>. It's just a bad practice, which somebody can follow later 😄 

Leave it if it's difficult to change.

What happens in more complicated returns is that React replaces that part of the array with the new one, but in the code it's still happens as `componentDid/WillUpdate`and if there is any usage of `refs` inside - they break.

Reply:
> It's just a bad practice, which somebody can follow later :smile:

Fair enough